src/accelerate/utils/bitsandbytes.py

```diff
@@ -0,0 +1,332 @@
+                                    "lora_dropout": lora_config.lora_dropout,
+                                    "fan_in_fan_out": lora_config.fan_in_fan_out,
+                                    "init_lora_weights": lora_config.init_lora_weights,
+                                }
+                                fourbit_kwargs = lora_kwargs.copy()
+                                fourbit_kwargs.update(
+                                    {
+                                        "compute_dtype": quantization_config.bnb_4bit_compute_dtype,
+                                        "compress_statistics": quantization_config.bnb_4bit_use_double_quant,
+                                        "quant_type": quantization_config.bnb_4bit_quant_type,
+                                    }
+                                )
+                                bnb_module = lora.Linear4bit(
+                                    adapter_name,
+                                    module.in_features,
+                                    module.out_features,
+                                    bias=module.bias is not None,
+                                    **fourbit_kwargs,
+                                )
+                                peft_module_found = True
+                                break
+                    if not peft_module_found:
+                        bnb_module = bnb.nn.Linear4bit(
+                            module.in_features,
+                            module.out_features,
+                            module.bias is not None,
+                            quantization_config.bnb_4bit_compute_dtype,
+                            compress_statistics=quantization_config.bnb_4bit_use_double_quant,
+                            quant_type=quantization_config.bnb_4bit_quant_type,
+                        )
+                    bnb_module.weight.data = module.weight.data.clone()
+                    if module.bias is not None:
+                        bnb_module.bias.data = module.bias.data.clone()
+                    setattr(model, name, bnb_module)
+                    has_been_replaced = True
+                # Force requires grad to False to avoid unexpected errors
+                module.requires_grad_(False)
+
+        if len(list(module.children())) > 0:
+            _, has_been_replaced = _replace_with_bnb_linear_with_weight(
+                module, modules_to_not_convert, current_key_name, quantization_config, peft_config
+            )
+        # Remove the last key for recursion
+        current_key_name.pop(-1)
+    return model, has_been_replaced
+
+
+def has_bnb_layers(model):
+    """Check if we have `bnb.nn.Linear4bit` or `bnb.nn.Linear8bitLt` layers inside our model"""
+    for m in model.modules():
+        if isinstance(m, bnb.nn.Linear4bit) or isinstance(m, bnb.nn.Linear8bitLt):
+            return True
+    return False
+
+
+def prepare_model_for_kbit_peft_training(model, use_gradient_checkpointing=True):
```

Comment:
```suggestion
def prepare_model_for_quantized_peft_training(model, use_gradient_checkpointing=True):
```
kbit does not mean anything since k can be equal to 32 and then it's normal training